scheduler/test/cook/test/mesos/mesos_mock.clj

```diff
@@ -0,0 +1,844 @@
+               {"e8ab3892-3023-4546-888d-67cc13eac3cc"
+                {:hostname "hosta"
+                 :attributes {}
+                 :resources {:cpus {"*" 10}
+                             :mem {"*" 100000}
+                             :ports {"*" [{:begin 10000 :end 20000}]}}
+                 :available-resources {:cpus {"*" 10}
+                                       :mem {"*" 100000}
+                                       :ports {"*" [{:begin 10000 :end 20000}]}}
+                 :slave-id "e8ab3892-3023-4546-888d-67cc13eac3cc"}
+
+                "a5cb3212-3023-4546-888d-67cc13eac3dd"
+                {:hostname "hostb"
+                 :attributes {}
+                 :resources {:cpus {"*" 12}
+                             :mem {"*" 120000}
+                             :ports {"*" [{:begin 10000 :end 30000}]}}
+                 :available-resources {:cpus {}
+                                       :mem {}
+                                       :ports {}}
+                 :slave-id "a5cb3212-3023-4546-888d-67cc13eac3dd"}}
+               :offer-id->offer
+               {"8fe5bf6c-ea39-4700-b1f6-9fd8a0271d7c"
+                {:id {:value "8fe5bf6c-ea39-4700-b1f6-9fd8a0271d7c"}
+                 :slave-id {:value "a5cb3212-3023-4546-888d-67cc13eac3dd"}
+                 :hostname "hostb"
+                 :resources [{:name "cpus" :role "*" :scalar 7 :type :value-scalar}
+                             {:name "mem" :role "*" :scalar 70000 :type :value-scalar}
+                             {:name "ports" :role "*" :ranges [{:begin 10000 :end 20000}] :type :value-ranges}]}
+                "45cb7812-1233-4546-888d-67cc13eac398"
+                {:id {:value "45cb7812-1233-4546-888d-67cc13eac398"}
+                 :slave-id {:value "a5cb3212-3023-4546-888d-67cc13eac3dd"}
+                 :hostname "hostb"
+                 :resources [{:name "cpus" :role "*" :scalar 5 :type :value-scalar}
+                             {:name "mem" :role "*" :scalar 50000 :type :value-scalar}
+                             {:name "ports" :role "*" :ranges [{:begin 20001 :end 30000}] :type :value-ranges}]}}
+               :task-id->task {}}
+        new-state (mm/handle-action! :launch {:tasks tasks :offer-ids offer-ids} state nil (noop-mock-scheduler))]
+    (is (= (count (:offer-id->offer new-state)) 0))
+    (is (= (count (:task-id->task new-state)) 1))
+    (is (= (get-in new-state [:slave-id->host "a5cb3212-3023-4546-888d-67cc13eac3dd"])
+           {:hostname "hostb"
+            :attributes {}
+            :resources {:cpus {"*" 12}
+                        :mem {"*" 120000}
+                        :ports {"*" [{:begin 10000 :end 30000}]}}
+            :available-resources {:cpus {"*" 4.0}
+                                  :mem {"*" 40000.0}
+                                  :ports {"*" [{:begin 10000 :end 12000} {:begin 25001 :end 30000}]}}
+            :slave-id "a5cb3212-3023-4546-888d-67cc13eac3dd"}))))
+
+(defn make-random-host
```

Comment:
`make-random-host` sort of implies that the contents of the host are random, when really it's only the `uuid` and `slave-id` that are. Maybe `dummy-host`?